### PR TITLE
add security property to WNM for access controlled data (#643)

### LINKED ISSUE #643: add security property to link in data-notification for access-controlled data
When we add an access-token to secure the data using 'wis2box auth add-token' , the wis2box-management container needs to add a security-property to the link of the data:

image

### PR DESCRIPTION
**Pull request by tomkralidis**
Fixes #643

### CODE REVIEW

File: wis2box-management/wis2box/pubsub/message.py [hunk 1]
```diff
@@ -210,6 +212,30 @@ def __init__(self, identifier: str, metadata_id: str, filepath: str,
...
+            record = oar.collection_item('discovery-metadata', metadata_id)
+
+            if record['wis2box'].get('has_auth'):
+                LOGGER.debug('Updating message with access control')
+
+                for link in self.message['links']:
+                    if link['href'] == public_file_url:
+                        LOGGER.debug('Adding security object to link')
+                        link['security'] = {
+                            'default': {
```
Comment (david-i-berry) [inline wis2box-management/wis2box/pubsub/message.py:227]: Is the use of `default `consistent with https://raw.githubusercontent.com/OAI/OpenAPI-Specification/3.1.0/schemas/v3.0/schema.yaml#/definitions/Components/properties/securitySchemes:~:text=HTTPSecurityScheme%3A%0A%20%20%20%20type%3A%20object,not%3A%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20enum%3A%20%5Bbearer%5D

Comment (tomkralidis) [reply wis2box-management/wis2box/pubsub/message.py:227]: Yes, one can define 1..n security objects, defined by user-specified keys.  `default` is a convention used by other centres (dk-dmi for example) and is mentioned in the [WIS2 Cookbook](https://wmo-im.github.io/wis2-cookbook/cookbook/wis2-cookbook-DRAFT.html#_publish_a_wis2_notification_message_with_access_control).

**Comment by david-i-berry** [comment]
OK, there's a difference in the schema at

https://wmo-im.github.io/wis2-notification-message/standard/wis2-notification-message-DRAFT.html

and

https://[schemas.wmo.int/wnm/1.0.0/schemas/wis2-notification-message-bundled.json](https://schemas.wmo.int/wnm/1.0.0/schemas/wis2-notification-message-bundled.json)

**Comment by maaikelimper** [comment]
Tested on test-server, with auth:
```
{
  "id": "6c787bd9-b2aa-479a-93c4-6d52272b75d1",
  "type": "Feature",
  "conformsTo": [
    "http://wis.wmo.int/spec/wnm/1/conf/core"
  ],
  "geometry": {
    "coordinates": [
      23.94046026,
      47.77706163
    ],
    "type": "Point"
  },
  "properties": {
    "data_id": "maaike-test:reco.surface-based-observations.synop/WIGOS_0-1234-5-WMOHQ_20240822T135600",
    "datetime": "2024-08-22T13:56:00Z",
    "pubtime": "2024-08-22T13:56:41Z",
    "integrity": {
      "method": "sha512",
      "value": "c2C1SYYBhP1Ly950nOv+OV2rKzQBQydxb83q6Lymt+IUeHAv2jD0/vx0Q/8JBvZBdtANUC4mOS1UAh99UGUcyQ=="
    },
    "metadata_id": "urn:wmo:md:maaike-test:reco.surface-based-observations.synop",
    "wigos_station_identifier": "0-1234-5-WMOHQ"
  },
  "links": [
    {
      "rel": "canonical",
      "type": "application/x-bufr",
      "href": "http://136.156.130.194/data/2024-08-22/wis/urn:wmo:md:maaike-test:reco.surface-based-observations.synop/WIGOS_0-1234-5-WMOHQ_20240822T135600.bufr4",
      "length": 384,
      "security": {
        "default": {
          "type": "http",
          "scheme": "bearer",
          "description": "Please contact the data provider for access"
        }
      }
    },
    {
      "rel": "via",
      "type": "text/html",
      "href": "https://oscar.wmo.int/surface/#/search/station/stationReportDetails/0-1234-5-WMOHQ"
    }
  ],
  "generated-by": "wis2box 1.0b7"
}
```
without auth:
```
{
  "id": "a2420338-53a5-4b04-a800-db47d91f9ed7",
  "type": "Feature",
  "conformsTo": [
    "http://wis.wmo.int/spec/wnm/1/conf/core"
  ],
  "geometry": {
    "coordinates": [
      23.94046026,
      47.77706163
    ],
    "type": "Point"
  },
  "properties": {
    "data_id": "maaike-test:reco.surface-based-observations.synop/WIGOS_0-1234-5-WMOHQ_20240822T140000",
    "datetime": "2024-08-22T14:00:00Z",
    "pubtime": "2024-08-22T14:00:20Z",
    "integrity": {
      "method": "sha512",
      "value": "n3xwI/A0eeBgSE+M9weM/sjnBGEwrk91XBWRJnGloix2LArEVzRzA8/z6n5P4POfB54kjIevJh2KyMXYWrHNcQ=="
    },
    "metadata_id": "urn:wmo:md:maaike-test:reco.surface-based-observations.synop",
    "content": {
      "encoding": "base64",
      "value": "QlVGUgABgAQAABYA/////wAAAAJuHgAH6AgWDgAAAAALAAABgMGWx2AAAVMAAE0gAFV01PSFEAAAAAAAAAAAAAAP////////////////////////////+P0Qs4BpHZUm5gfCNGEap///////////////////////////+duqNP/////+CZAB9P/3R3cw+h////////////////////////////////////////////////8wiAAX//////////Af////gP////////////////////////////////////+AyP//////////////////////////+J/YPAPff2DwGT4goBadMCgN3//////////////////////////////////////////A+j/f/AMH/QDZ/oBQf0AYH6AH3////////////////////////////////////////////////////////////////////////////////////8A3Nzc3",
      "size": 384
    },
    "wigos_station_identifier": "0-1234-5-WMOHQ"
  },
  "links": [
    {
      "rel": "canonical",
      "type": "application/x-bufr",
      "href": "http://136.156.130.194/data/2024-08-22/wis/urn:wmo:md:maaike-test:reco.surface-based-observations.synop/WIGOS_0-1234-5-WMOHQ_20240822T140000.bufr4",
      "length": 384
    },
    {
      "rel": "via",
      "type": "text/html",
      "href": "https://oscar.wmo.int/surface/#/search/station/stationReportDetails/0-1234-5-WMOHQ"
    }
  ],
  "generated-by": "wis2box 1.0b7"
}
```

**Review by maaikelimper — APPROVED** [review]
Tested on test-server and it seems to work correctly, approved

**Pull request merged**